Language/Java/Syntax.hs

```diff
@@ -995,7 +995,7 @@ deriving instance ReadExtension p => Read (Exp p)
 deriving instance DataExtension p => Data (Exp p)
 
 instance EqualityExtension p => Equality (Exp p) where
-  eq opt (Lit l1) (Lit l2) =
+  eq opt (Lit _ l1) (Lit _ l2) =
     eq opt l1 l2
```

Comment:
Hier sollte korrekterweise nicht einfach ein `_` verwendet werden. Stattdessen soll hier wie auch an vergleichbaren Stellen für die SourceSpans auch ein Vergleich aufgerufen werden mit `eq opt s1 s2`

Reply:
Außerdem fehlt noch in der Instanz von `Located` die neue Definition für die `sourceSpan`-Funktion, mit der die SourceSpan einer `Exp` erlangt werden kann.